New input format (#5)
hdr-bgnn/Crop_image: README.md

## 1 - Short description

>   + We use raw fish image [available at Tulane](http://www.tubri.org/HDR/INHS/) and Metadata json file generated by [Drexel_metadata_rformatter] (https://github.com/hdr-bgnn/drexel_metadata_formatter) to crop the fish out of a more complexe image.

Reviewer: I don't think the reformatter generates the metadata JSON, it only reformats it. Isn't that so?

Author: you are right.
I change the text  to
We use raw fish image [available at Tulane](http://www.tubri.org/HDR/INHS/) and Metadata json file generated by [Drexel_metadata](https://github.com/hdr-bgnn/drexel_metadata) and reformatted by [Drexel_metadata_formatter](https://github.com/hdr-bgnn/drexel_metadata_formatter) to crop the fish out of a more complexe image.

>   + The metadata json file contained the bounding box of the fish which use to crop the fish [here is an example](Test_data/INHS_FISH_000742_cropped.jpg). We increase the size of the bounding box around the fish by 5% increase (by default, parameter) in case the bounding box (from the metadata) would cut the edge of the fish.

Reviewer: ```suggestion
  + The metadata json file contained the bounding box of the fish which use to crop the fish [here is an example](Test_data/INHS_FISH_000742_cropped.jpg). We increase the size of the bounding box around the fish by default by 5%  (see `--increase` parameter), to reduce the chance that the bounding box (from the metadata) would cut the edge of the fish.
```

**input**
![original fish image](Test_data/INHS_FISH_000742.jpg)

**Cropped output**
![Cropped fish image](Test_data/INHS_FISH_000742_cropped.jpg)

## 2 - Usage

Input example : 
  + fish image .jpg [see INHS_FISH_000742.jpg](Test_data/INHS_FISH_000742.jpg)
  + Metadata.json [see INHS_FISH_000742.json](Test_data/INHS_FISH_000742.json)
  
Ouput :
  + Cropped image .jpg [see INHS_FISH_000742_cropped.jpg]Test_data/INHS_FISH_000742_cropped.jpg)

Usage in python
you need the environment define by crop_env.yml. I suggest to use anaconda or miniconda as environment manager

```
Crop_image_main.py [-h] [--increase] input_image imput_metadata output
```

Example using the Test_data with 10% increase in size

```
Crop_image_main.py INHS_FISH_000742.jpg INHS_FISH_000742.json INHS_FISH_000742_cropped.jpg --increase 0.10
```

> The `increase` parameter defines the percentage increase of the bounding box from the original metadata size in percentage for each direction (in this example 10% increase in width and in height is written as 0.10). The default value in 0.05.

Reviewer: ```suggestion
The `--increase` parameter defines the fractional increase of the bounding box from the original metadata size in each direction (10% in this example). The default value in 0.05.
```

## 2 - Usage

Input example : 
  + fish image .jpg [see INHS_FISH_000742.jpg](https://github.com/hdr-bgnn/Crop_image/blob/main/Test_data/INHS_FISH_000742.jpg)

>   + Metadata.json [see INHS_FISH_000742.json](https://github.com/hdr-bgnn/Crop_image/blob/new-input-format/Test_data/INHS_FISH_000742.json)

Reviewer 2: The above link contains an URL to the JSON file in the `new-input-format` branch.
Once this PR is merged and this branch is deleted the link will no longer work.
You can just provide a relative path that will always point to the file in the current branch.

```suggestion
  + Metadata.json [see INHS_FISH_000742.json](Test_data/INHS_FISH_000742.json)
```

Reviewer: @thibaulttabarin I think the request here was to change _all_ such paths accordingly, not just this one.

Author: Chnaged to relative path
https://github.com/hdr-bgnn/Crop_image/commit/ae3c32f9620f25c0da6b4c1d3540e4cf95f0e23c

you need the environment define by crop_env.yml. I suggest to use anaconda or miniconda as environment manager

```
Crop_image_main.py [-h] [--increase] input_image imput_metadata output
```

Example using the Test_data with 5% increase in size

```
Crop_image_main.py INHS_FISH_000742.jpg INHS_FISH_000742.json INHS_FISH_000742_cropped.jpg --increase 0.05
```

> The `increase` parameter defines the percentage increase of the bounding box from the original metadata size in percentage for each direction (5% increase in widht and 5% in height as 0.05)

Reviewer 2: ```suggestion
The `increase` parameter defines the percentage increase of the bounding box from the original metadata size in percentage for each direction (5% increase in width and 5% in height as 0.05)
```

Reviewer 2: Why not show an example with a different increase amount since `0.05` is the default?

Reviewer 2: I would also add a comment answering this question from the issue:
```
Also, when calculating the pixels to add in each direction, are fractions subject to floor or ceiling?
```
From what I can tell the answer is floor since we are using `int()` which truncates towards zero:
https://github.com/hdr-bgnn/Crop_image/blob/e52c367dde6cce08c4accdd0f4f92ce4eb04e47d/Crop_image_main.py#L79-L80

Author: yes it is floor(). May be we could use round() which would convert to the closest integer, but ultimately this is minor since in this case or the other we lose or gain a pixel or two which won't change much the cropping.

Reviewer: @thibaulttabarin I don't think you've addressed this yet. Specifically, the following remains unaddressed (and would be trivial to implement):
> Why not show an example with a different increase amount since 0.05 is the default?

Also, another comment suggests to document the fact that the calculation of the number of pixels to add is subject to integer truncation (which, BTW, strictly speaking [is _not_ the same as `floor()`](https://en.wikipedia.org/wiki/Floor_and_ceiling_functions): floor(-2.4) == 3, including in Python; that said, for positive integers it is the same).

Author: Change the increase value in the documentation example 
https://github.com/hdr-bgnn/Crop_image/pull/5/commits/b53cbd2c629c22563d8bf96d81fada235deb3652
